source/concurrent/Promise.ooc

```diff
@@ -19,7 +19,19 @@ _Synchronizer: abstract class {
 	_state: _PromiseState
 	init: func
 	wait: abstract func (time: TimeSpan) -> Bool
-	wait: func ~forever -> Bool { this wait(TimeSpan maximumValue) }
+	wait: func ~forever -> Bool {
+		version(debugDeadlock) {
+			timeLimitTicks := 2_000
```

Comment:
Agree on ``Timer`` returning a `TimeSpan`. The problem though of using "real time" for the `timer` in this case is that `_Task wait(TimeSpan)` uses ticks. 

```ooc
timer := WallTimer new() . start()
result := this wait(TimeSpan milliseconds(1000)) 
endTime := timer stop() 
```
``endTime`` in that case will probably be < 1000 since a cpu tick is probably counted faster than a "real" millisecond.